api/v1alpha1/storageclassclaim_types.go

```diff
@@ -73,6 +73,8 @@ type StorageClassClaim struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="oldSelf == self",message="spec is immutable"
```

Comment:
please push the generated CRD files as this validation need to be added to CRD yaml